src/main/java/rainy/tasks/TaskTracker.java

```diff
@@ -0,0 +1,221 @@
+package rainy.tasks;
+
+import java.util.ArrayList;
+import java.util.Collections;
+
+import rainy.database.UI;
+import rainy.rainyexceptions.InvalidIndexException;
+import rainy.rainyexceptions.InvalidMarkAndUnmarkException;
+
+/**
+ * Keeps track of the list of classes and variables pertaining to how many tasks there are.
+ *
+ */
+public class TaskTracker {
+    private ArrayList<Task> taskList;
+    private int counter;
+    private boolean receivedInputs;
```

Comment:
Perhaps you could rename this such that it sounds more like a boolean?

maybe `isInputReceived` or `hasReceivedInput`?

Reply:
Thank you for the suggestion :)